linkerd/proxy/api-resolve/src/metadata.rs

```diff
@@ -18,7 +19,7 @@ pub struct Metadata {
     tagged_transport_port: Option<u16>,
 
     /// How to verify TLS for the endpoint.
-    identity: Option<ServerId>,
+    identity: Option<(ServerId, ServerName)>,
```

Comment:
Can we perhaps change this to store a [`tls::client::ClientTls`](https://github.com/linkerd/linkerd2-proxy/blob/b786b5006e6f8b78a3c96cd4a2aaef8c799b80e7/linkerd/tls/src/client.rs#L22-L28)? This also opens the option to pass through ALPN settings (later).

Reply:
This could also potentially make sense as a discrete change to help minimize the diff we need to consider for URI handling.